.github/workflows/runtest-win.yml

```diff
@@ -37,13 +42,33 @@ jobs:
         run: |
           choco install --yes --no-progress dmd winflexbison3
 
-      - name: runtest
+      - name: runtest (optenv=${{ matrix.optenv }})
+        if: ${{ !matrix.optenv }}
         run: |
           python runtest.py --all --exclude-list=windows_ci_skip.txt --time --jobs=4
 
-      - name: Archive Failed tests
+      - name: runtest (optenv=${{ matrix.optenv }})
+        if: ${{ matrix.optenv }}
+        run: |
+          python runtest.py --all --exclude-list=windows_ci_skip.txt --time --jobs=4 --faillog=failed_tests_optenv.log
```

Comment:
I think we could vastly shrink the # of tests run in this part of the logic. In theory a single MSVC test should be sufficient.  If it's not broken then we've not changed the logic to not work when the below env vars are set..

Thoughts?
BTW we do have a limit on our GHA usage, which is complicated to calculate, but not unlimited, so 2x'ing the windows gha usage is not worth it in this case.

Reply:
> I think we could vastly shrink the # of tests run in this part of the logic. In theory a single MSVC test should be sufficient. If it's not broken then we've not changed the logic to not work when the below env vars are set..
>
> Thoughts?

I think that is a really good idea.

Locally, I usually run the test suite with a file that only contains the msvc/msvs tests.  I think there are approximately 80 tests.  About 24-ish won't be run since they are for VS2013 and earlier.

As mentioned above, a subdirectory off the SCons root for test runner files would be a good idea.  Perhaps something like an `.scons` subdirectory.

I think the exclude files and limited test files are useful outside of GHA tests and perhaps deserve their own home.

Reply:
If we limit the "optional environment" tests based on an input test file, it should probably be in a standalone workflow file.

"Full Test Suite on Windows" would not be accurate any more.  "MSVC Test Suite on Windows" might be a better stand-alone workflow file.

I don't think there would be any changes to `runtest-win.yml` then.

I'm not sure if I add a new workflow that it will run since it doesn't exist already in the master branch.

Reply:
How about the tests to run and/or exclude added to `testing/ci`, with current naming pattern?

Reply:
Updated per discussion.